File: 119.pascals-triangle-ii.js
// @leet start
/**
 * @param {number} rowIndex
 * @return {number[]}
 */
var getRow = function(rowIndex) {
  let row = [];
  for (let i = 0; i <= rowIndex; i++) {
    let prev = []
    for (let j = 0; j <= i; j++) {
      if (j === 0 || j === i) {
        prev.push(1)
      } else {
        prev.push(row[j - 1] + row[j])
      }
    }
    row = prev
  }
  return row
};
// @leet end

File: 119.pascals-triangle-ii.go
// @leet start
func getRow(rowIndex int) []int {
  res := []int{1}
  for i := 0; i <= rowIndex; i++ {
    row := make([]int, i+1)
    for j := 0; j <= i; j++ {
      if j == 0 || j == i {
        row[j] = 1
      } else {
        row[j] = res[j-1] + res[j]
      }
    }
    res = row
  }
  return res
}
// @leet end
